### PR TITLE
Add components.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: components/echo/echo.c [hunk 1]
```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2024 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the tests/LICENSE file.
```
Comment (kasperl) [inline components/echo/echo.c:3]:
```suggestion
// be found in the LICENSE file.
```

File: components/echo/echo.toit [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2024 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the tests/LICENSE file.
```
Comment (kasperl) [inline components/echo/echo.toit:3]:
```suggestion
// be found in the LICENSE file.
```

File: components/echo/echo.toit [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+*/
+
+import system.external
+
+main:
+  echo := external.Client.open "toitlang.org/demo-echo"
+  echo.set-on-notify:: print "Got message: $it.to-string"
+  echo.notify "Hello, world!"
+
+  function-id := 0
```
Comment (kasperl) [inline components/echo/echo.toit:16]: Maybe pull this out as a constant? That feels simpler than this way of naming the argument.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: components/echo/echo.c (hunk 1 above)
Comment (floitsch) [reply components/echo/echo.c]: done.

File: components/echo/echo.toit (hunk 2 above)
Comment (floitsch) [reply components/echo/echo.toit]: done.

File: components/echo/echo.toit (hunk 3 above)
Comment (floitsch) [reply components/echo/echo.toit]: done.

**Pull request merged**